backend/logic/courses/courses.js

```diff
@@ -261,12 +261,108 @@ var updateCourse = async (course, newTitle, newTags) => {
   }
 };
 
+/**
+ * Assign a course to a curriculum
+ * @author Yufei Liu
+ * @param {string} curriculumType,{string} courseCode,{string} curriculum
+ * @returns true if successful
+ * @throws error if curriculumType, courseCode or curriculum dose not exist
+ *         course already assigned a curriculum
+ *
+ */
+var assignCourseToCurriculum = async (
+  curriculumType,
+  courseCode,
+  curriculum
+) => {
+  if (!format.isMcGillCourse(courseCode)) {
+    throw Error("Invalid course format!\n");
+  }
+
+  let conn = await mysql.getNewConnection();
+
+  const checkExistQuery = `SELECT COUNT(*) AS count
+  FROM (
+      SELECT course_code, curriculum_name FROM curriculum_tech_comps
+      union all
+      SELECT course_code, curriculum_name FROM curriculum_complementaries
+      union all
+      SELECT course_code, curriculum_name FROM curriculum_core_classes
+  ) a
```

Comment:
is the a after the parenthesis a typo? i don't recognize such syntax and i didn't find anything online that corresponds to this

Reply:
No it's not a typo, it's an alias, I think it's needed when you have nested statements. @si-x-li Correct me if I'm wrong

Reply:
It's equivalent to saying AS a

Reply:
didn't know about that!